packages/graphiql-2-rfc-context/src/components/VariableEditor.tsx

```diff
@@ -89,6 +92,7 @@ export function VariableEditor(props: VariableEditorProps) {
       if (!ignoreChangeEvent) {
         cachedValueRef.current = editor.getValue();
         session.changeVariables(cachedValueRef.current);
+        browser.changeQueryStringParams('variables', cachedValueRef.current);
```

Comment:
@acao question: should the `browser` context util be separated from the `session` context util? From what I can see, we seem to also be setting `session` whenever we are changing the query string. So perhaps the `session` context util can simply update the browser context as well? Or is there another reason why we have them separate that I'm not aware of?

Reply:
Session here refers to the current tab, incidentally haha. Maybe EditorSession would be more explicit?

Reply:
either way, all of this will be replaced. excellent points to consider in the next state management iteration.